### PR TITLE
ADAPT-4099: Upgrade storyblok to v2 bridge

### PR DESCRIPTION
**Pull request by Lyafel**
# Summary
- Updated to storybook v2

# Review By (Date)
- End of the sprint

# Criticality
- Low

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Follow the steps in the readme to get a localhost with https up and running
2. Navigate to storybook (https://app.storyblok.com/beta-v2/#/me/spaces/122401/stories/0/0/index/0)
4. Navigate to any content page and edit
5. Click around on the new interface and try out a few of the new controls

### CODE REVIEW

**Comment by Lyafel** [comment]
The new version doesn't accept http so there are some steps to follow to fix the local. The example used port 3010 so that's what I used for the test url. Is there a more permanent way to use https without that fix?

**Comment by sherakama** [comment]
@Lyafel I am confused. What doesn't support http?

**Comment by sherakama** [comment]
From the documentation at: https://www.storyblok.com/docs/guide/essentials/visual-editor

> HINT: When using http for your Preview URLs the Storyblok UI will redirect to a non https version as well, otherwise Browser Security restrictions won’t allow us to embed your site. This is only recommended for local development domains such as http://localhost or similar. The API requests to the Management API will still be done using https.

It appears that they did not change HTTP vs HTTPS support.

**Comment by Lyafel** [comment]
Screen Shot 2021-12-06 at 2 28 51 PM

This is the issue I was asking about

**Comment by sherakama** [comment]
@Lyafel Did you try method #2? It looks like you need to add a global variable `STORYBLOK_PREVIEW_URL` to make it work. Was that method successful?

**Comment by Lyafel** [comment]
@sherakama I did try and had no luck with making it work. I can try again, not sure if I placed the file in a wrong place or something else.

**Comment by sherakama** [comment]
Hmmm.

It appears they are breaking the current process with the requirement of https if you are going to use a localhost URL in their visual editor on their site. Method #2 appears to be a way to host your own storyblok app interface locally. I was able to make some progress by creating a file called `dev.html` in `/static/dev.html` and adding in the contents they provided. I was able to navigate around a V2 instance locally but I wasn't able to get the visual editor to work and had a number of console error logs.

@Lyafel can you look into path #1 for us?

1. Please create a npm script that will get us up and running with an https proxy. You can commit the local development cert and key to the repo for this and use the ports 3000 and 3010. You can add the new preview URL to the dev space in storyblok. 

It would be a "nice to have" if you could also look into version #2 where we could have our local dev environment up and running under HTTP as `netlify dev` runs under that and is a good way for us to run our LAMBDA functions locally. 

cc: @mattanglin @yvonnetangsu @rebeccahongsf @nplowman

File: src/pages/editor.js [hunk 1]
```diff
@@ -30,48 +33,45 @@ const getParam = function (val) {
  * This is Sparta
  */
 const initBridge = function (key, sbResolveRelations, setStory) {
+  const { StoryblokBridge } = window;
+
   // Initialize the Storyblok JS Bridge
-  window.storyblok.init({
+  const storyblokInstance = new StoryblokBridge({
+    preventClicks: true,
```
Comment (sherakama) [inline src/pages/editor.js:40]: The preventClicks option doesn't seem to work for me. I still navigate around in the editor and I am not sure why.

**Comment by sherakama** [comment]
@yvonnetangsu and @rebeccahongsf please review.

File: src/pages/editor.js (hunk 1 above)
Comment (Lyafel) [reply src/pages/editor.js:40]: I tried to investigate this but no luck so far. Should we open a separate ticket or do you want me to dig some more into this now?

Comment (sherakama) [reply src/pages/editor.js:40]: Thanks for looking into it. I'll add a ticket to our backlog to follow up with Storyblok.

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu and @rebeccahongsf please review.

Sorry for the delay. I'll pull this down and review for real tomorrow 😂

**Comment by yvonnetangsu** [comment]
@Lyafel @sherakama I followed the readme and it did work for me after I pop the 3010 out into its own tab and click proceed. I have tried the control and it seems to work as well. I'm assuming that the update in this PR will work for the current v1 version, and it should work for v2 when the official release comes out next year?

Unrelated to this PR but: Looks like Storyblok still has a lot to do. I haven't figured out how to add or modify presets yet 😬  Looks like the whole Components page from the left sidebar isn't there either.

File: README.md [hunk 2]
```diff
@@ -39,17 +39,37 @@ Connecting to Storyblok
...
+---
+
+Currently Storyblok v2 doesn't allow accessing the environment with http, so to make it work, https has to be added to localhost. For your convenience we have created the certificate and key but your system may not trust this self signed cert. Once you have localhost up and running you can visit the url in the browser and proceed past the warning or you can add the certificate to your trusted list.
+
+OSX:
+https://readwriteexercise.com/posts/trust-self-signed-certificates-macos/
+
+To start:
+```
+npm run dev
```
Comment (yvonnetangsu) [inline README.md:52]:
```suggestion
npm run develop
```
I think we have `develop` but not `dev` 😄

File: README.md [hunk 3]
```diff
@@ -39,17 +39,37 @@ Connecting to Storyblok
...
+npm run https-proxy-start
```
Comment (yvonnetangsu) [inline README.md:53]: Could we mention do this in a separate terminal?

**Comment by sherakama** [comment]
> Are we actually going to merge this into production before v2 is ready or wait and just have this in dev or leave it open?

https://www.storyblok.com/v2
The visual editor part of the v2 upgrade is "done" but you are right on the rest of their UI. 

Even though this is compatible with the V1, let's hold on this until Storyblok does a full release of their v2 as there is a lot for them to finish and I'd like to stay away from having our clients use it until it is really ready.

**Comment by yvonnetangsu** [comment]
> > Are we actually going to merge this into production before v2 is ready or wait and just have this in dev or leave it open?
> 
> https://www.storyblok.com/v2 The visual editor part of the v2 upgrade is "done" but you are right on the rest of their UI.
> 
> Even though this is compatible with the V1, let's hold on this until Storyblok does a full release of their v2 as there is a lot for them to finish and I'd like to stay away from having our clients use it until it is really ready.

Agree! 😄

**Review by sherakama — APPROVED** [review]
GTG

**Pull request merged**